tekton/README.md

```diff
@@ -34,7 +38,19 @@ and
 
 ## Release Pipeline
 
-The `Tasks` which make up our release `Pipeline` are:
+The release pipeline uses the
+[`golang`](https://github.com/tektoncd/catalog/tree/master/golang)
+Tasks from the
+[`tektoncd/catalog`](https://github.com/tektoncd/catalog). To add them
+to your cluster:
+
+```
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/golang/lint.yaml
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/golang/build.yaml
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/golang/tests.yaml
```

Comment:
We can keep discussing this (I don't want to block merging this!) but I'm not totally sold on running all the tests as part of the release pipeline. The tests are run on the code merged with HEAD before merging, so if one of these tests fails it would mean that either 1) something we are not accounting for changed (e.g. a dependency that's being pulled in) or 2) something is flakey. I think 99.9% of the time it would be the second case, in which case the tests would just slow down the release process + result in having to run it again. I guess we would be doing this to try to catch (1)? I would rather try to catch that by running continuous tests than doing it at release time 🤔 

What are your thoughts on this @vdemeester ? (Maybe this would catch something else I haven't thought of - or maybe it's just worth it for the increased confidence!)

Reply:
So overall I do agree with you *but* there is one case that is not clear yet. I'll took 0.5.2 to explain this one — as we cherry-picked commits on the branch, we didn't do pull-request, and thus we did not run the entire test suites on it — that's mainly why I did https://github.com/tektoncd/pipeline/pull/1088.

That's why I felt safe to run tests during the release (even if, there, it's only the unit tests :sweat_smile:). 

Honestly I am ok not running them but I do like the idea that if it fails because of flakyness:
- it will make us prioritize fixing that flakyness :stuck_out_tongue_closed_eyes: 
- it slows the release but by a tiny tiny amount of time (a matter of minutes :stuck_out_tongue_closed_eyes:)

Reply:
For me the main point of doing release time testing is if we do extra testing there that is normally too time/resource consuming to run on each commit/PR - I'm not sure we have such tests (yet?) though. I guess we need to see how long the upgrade testing will take to run once it's been written.

Another way to run those kind of tests is to have a periodic (daily?) run of those tests; the issue I've seen in the past with periodic tests is that is hard to have people putting effort in keeping them green - since they don't block merging patches - but if we use such tests for pre-release it might give the extra motivation needed.

Kind of diverging from the main topic, periodic testing, if kept green, have an extra value, which is to provide a source of "clean" data to detect flaky test. Test jobs on PR may fail because the code in the PR makes them fail, however periodic tests should be 100% passing, so failures there are good data for identifying and troubleshooting flaky tests.